core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/rds/RdsConnectionValidator.java

```diff
@@ -1,26 +1,62 @@
 package com.sequenceiq.cloudbreak.controller.validation.rds;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
 import java.sql.Connection;
-import java.sql.DriverManager;
+import java.sql.Driver;
 import java.sql.SQLException;
 import java.util.Properties;
 
+import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
+import com.sequenceiq.cloudbreak.api.model.DatabaseVendor;
 import com.sequenceiq.cloudbreak.controller.BadRequestException;
+import com.sequenceiq.cloudbreak.domain.RDSConfig;
 
 @Component
 public class RdsConnectionValidator {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(RdsConnectionValidator.class);
 
-    public void validateRdsConnection(String connectionURL, String connectionUserName, String connectionPassword) {
+    public void validateRdsConnection(RDSConfig rdsConfig) {
+        if (StringUtils.isEmpty(rdsConfig.getConnectorJarUrl())) {
+            if (rdsConfig.getDatabaseEngine().equals(DatabaseVendor.POSTGRES.name())) {
+                validateRdsConnection(new org.postgresql.Driver(), rdsConfig);
+                return;
+            } else {
+                throw new BadRequestException("Cannot validate connection without a jdbc connector.");
+            }
+        }
+
+        try {
+            URL urls[] = {new URL("jar:" + rdsConfig.getConnectorJarUrl() + "!/")};
+            try (URLClassLoader cl = URLClassLoader.newInstance(urls)) {
+                DatabaseVendor vendor = DatabaseVendor.valueOf(rdsConfig.getDatabaseEngine());
+                cl.loadClass(vendor.connectionDriver());
+                Driver d = (Driver) cl.loadClass(vendor.connectionDriver()).newInstance();
+                validateRdsConnection(d, rdsConfig);
+            } catch (Exception e) {
+                throw new BadRequestException("Failed to connect to RDS: " + e.getMessage(), e);
```

Comment:
We should handle when the connector JAR is not reachable, because it could be available only from the cluster's network and differentiate the case when connector jar available but not able to connect to the RDS instance due to network or auth issue.

Also this functionality could be covered by unit tests.